app/models/mixins/process_tasks_mixin.rb

```diff
@@ -11,16 +11,20 @@ def process_tasks(options)
         task_audit_event(:success, options, :message => msg)
       else
         assert_known_task(options)
-        options[:userid] ||= "system"
+        options[:userid] = User.current_user.try(:userid) || "system"
```

Comment:
@lfu 
Won't this override the options[:userid] if it exists? Backward compatibility